front/pages/api/w/[wId]/data_sources/managed.ts

```diff
@@ -328,9 +333,35 @@ async function handler(
         });
       }
 
-      const vault = await (provider === "webcrawler"
-        ? VaultResource.fetchWorkspaceGlobalVault(auth)
-        : VaultResource.fetchWorkspaceSystemVault(auth));
+      let vault: VaultResource | null = null;
```

Comment:
I'd be in favor of creating a new dedicated endpoint to create data source in a vault. This endpoint should probably only be used for proper managed data sources (e.g: Slack, Notion, ...).

Reply:
:+1: bullish to go through something under the vault path in the URL